### PR TITLE
fix(rust): `LazyFrame::join()` no longer ignores 3 `JoinArgs` parameters

### PR DESCRIPTION
**Pull request by Bromeon**
The `JoinArgs` fields `validation`, `join_nulls` and `suffix` were not considered before.

The builder currently has no option for the `slice` parameter, so I didn't add it, but wrote a comment.

### CODE REVIEW

File: crates/polars-lazy/src/frame/mod.rs [hunk 1]
```diff
@@ -1225,12 +1225,23 @@ impl LazyFrame {
...
             .how(args.how)
-            .finish()
+            .validate(args.validation)
+            .join_nulls(args.join_nulls);
+
+        if let Some(suffix) = args.suffix {
+            builder = builder.suffix(suffix);
+        }
+
+        // Note: args.slice is currently ignored.
```
Comment (ritchie46) [inline crates/polars-lazy/src/frame/mod.rs:1242]:
```suggestion
        // Note: args.slice is set by the optimizer
```

**Comment by Bromeon** [comment]
> // Note: args.slice is set by the optimizer

Good to know! Should we add this to the method docs as well?

**Comment by ritchie46** [comment]
We can.

**Review by ritchie46 — APPROVED** [review]
Thank you @Bromeon

**Pull request merged**